spec/controllers/invite_requests_controller_spec.rb

```diff
@@ -0,0 +1,188 @@
+
+describe InviteRequestsController do
+  include LoginMacros
+  include RedirectExpectationHelper
+  let(:admin) { create(:admin) }
+  let(:user) { create(:user) }
+
+  describe "GET #index" do
+    it "renders" do
+      get :index
+      expect(response).to render_template("index")
+      expect(assigns(:invite_request)).to be_a_new(InviteRequest)
+    end
+  end
+
+  describe "GET #show" do
+    context "given invalid emails" do
+      it "redirects to index with error" do
+        message = "You can search for the email address you signed up with below. If you can't find it, your invitation may have already been emailed to that address; please check your email Spam folder as your spam filters may have placed it there."
+        get :show
+        it_redirects_to_with_error(invite_requests_path, message)
+        expect(assigns(:invite_request)).to be_nil
+        get :show, email: "mistressofallevil@example.org"
+        it_redirects_to_with_error(invite_requests_path, message)
+        expect(assigns(:invite_request)).to be_nil
+      end
+
+      it "renders for an ajax call" do
+        xhr :get, :show
+        expect(response).to render_template("show")
+        expect(assigns(:invite_request)).to be_nil
+        xhr :get, :show, email: "mistressofallevil@example.org"
+        expect(response).to render_template("show")
+        expect(assigns(:invite_request)).to be_nil
+      end
+    end
+
+    context "given valid emails" do
+      let(:invite_request) { create(:invite_request) }
+
+      it "renders" do
+        get :show, email: invite_request.email
+        expect(response).to render_template("show")
+        expect(assigns(:invite_request)).to eq(invite_request)
+      end
+
+      it "renders for an ajax call" do
+        xhr :get, :show, email: invite_request.email
+        expect(response).to render_template("show")
+        expect(assigns(:invite_request)).to eq(invite_request)
+      end
+    end
+  end
+
+  describe "POST #create" do
+    it "redirects to index with error given invalid emails" do
+      post :create, invite_request: { email: "wat" }
+      expect(response).to render_template("index")
+    end
+
+    it "redirects to index with error given valid emails" do
+      email = generate(:email)
+      post :create, invite_request: { email: email }
+      invite_request = InviteRequest.find_by_email(email)
+      it_redirects_to_with_notice(invite_requests_path, "You've been added to our queue! Yay! We estimate that you'll receive an invitation around #{invite_request.proposed_fill_date}. We strongly recommend that you add do-not-reply@archiveofourown.org to your address book to prevent the invitation email from getting blocked as spam by your email provider.")
+    end
+  end
+
+  describe "DELETE #destroy" do
+    it "blocks non-admins" do
+      delete :destroy
+      it_redirects_to_with_notice(root_path, "I'm sorry, only an admin can look at that area")
+
+      fake_login_known_user(user)
```

Comment:
I just noticed that I could have used `fake_login` instead, since we don't need to do anything with the user in this test.